io_scene_godot/converters/animation.py

```diff
@@ -298,22 +411,213 @@ def to_matrix(self):
+    if blender_lamp.animation_data is None:
+        return
+
+    first_frame, last_frame = get_action_frame_range(action)
+    base_node_path = NodePath(
+        animation_player.parent.get_path(), light_node.get_path()
+    )
+
+    animation_resource.add_simple_value_track(
+        action.fcurves, 'use_negative',
+        partial(
+            build_const_interp_value_track,
+            base_node_path.new_copy('light_negative'),
+            lambda x: x > 0.0,
+        )
+    )
+
+    for item in light_node.attribute_conversion:
+        bl_attr, gd_attr, converter, val_type = item
+        # vector vallue animation need special treatment
+        if val_type in (float, bool):
+            if val_type == float:
+                track_builder = build_linear_interp_value_track
+            else:
+                track_builder = build_const_interp_value_track
+            animation_resource.add_simple_value_track(
+                action.fcurves,
+                bl_attr,
+                partial(
+                    track_builder,
+                    base_node_path.new_copy(gd_attr),
+                    converter
+                )
+            )
+
+    # color tracks is not one-one mapping to fcurve, they
+    # need to be treated like transform track
+    color_frame_values_map = collections.OrderedDict()
+
+    for fcurve in action.fcurves:
+        _, attribute = split_fcurve_data_path(fcurve.data_path)
+
+        if attribute in ('color', 'shadow_color'):
+            if attribute not in color_frame_values_map:
+                color_frame_values_map[attribute] = [
+                    mathutils.Color()
+                    for _ in range(first_frame, last_frame)
+                ]
+            color_list = color_frame_values_map[attribute]
+            for frame in range(first_frame, last_frame):
+                color_list[frame - first_frame][
+                    fcurve.array_index] = fcurve.evaluate(frame)
+
+    for attribute, frame_value_list in color_frame_values_map.items():
+        if attribute == 'color':
+            track_path = base_node_path.new_copy('light_color')
+        else:
+            track_path = base_node_path.new_copy('shadow_color')
+
+        animation_resource.add_track(
+            Track(
+                'value',
+                track_path,
+                range(first_frame, last_frame),
+                frame_value_list
+            )
+        )
+
+
+def export_camera_action(camera_node, animation_player,
+                         blender_cam, action, animation_resource):
+    """Export camera action"""
+    # pylint: disable-msg=too-many-locals
+    # pylint: disable-msg=too-many-branches
+    if blender_cam.animation_data is None:
+        return
+
+    first_frame, last_frame = get_action_frame_range(action)
+    base_node_path = NodePath(
+        animation_player.parent.get_path(), camera_node.get_path()
+    )
+
+    for item in camera_node.attribute_conversion:
```

Comment:
I wonder if this for loop and the similar block in export_light_action can be made into a separate function?